sqlgen/db_test.go

```diff
@@ -123,6 +123,49 @@ func TestShardLimitInsert(t *testing.T) {
 	}
 }
 
+func TestShardLimitBatchInsert(t *testing.T) {
+	testcases := getLimitTestcases(t, "BatchInsert")
+	for _, testcase := range testcases {
+		t.Run(testcase.title, func(t *testing.T) {
+			tdb, db, err := setup()
+			assert.NoError(t, err)
+			defer tdb.Close()
+			ctx := context.Background()
+
+			aliceDb := testcase.withFilterLimit(db, Filter{
+				"name": "Alice",
+			})
+
+			alice := &User{Name: "Alice"}
+			bob := &User{Name: "Bob"}
+
+			// Check aliceDb can insert alice.
+			err = aliceDb.InsertRows(ctx, [](*User){alice}, 100)
+			assert.NoError(t, err)
+
+			// Check aliceDb can't insert alice and bob.
+			err = aliceDb.InsertRows(ctx, [](*User){alice, bob}, 100)
+			assert.Contains(t, err.Error(), "db requies name = Alice, but query has name = Bob")
+
+			tom := &User{Name: "Tom"}
+			jerry := &User{Name: "Jerry"}
+			// Check db can insert tom and jerry with a batch size of 1.
+			err = db.InsertRows(ctx, [](*User){tom, jerry}, 1)
+			assert.NoError(t, err)
```

Comment:
@changpingc here. Weird that github wasn't showing this previously. I forced pushed again

Reply:
can you check that these rows are inserted? this test would have passed without your fix for small batch.

Reply:
@changpingc modified the test